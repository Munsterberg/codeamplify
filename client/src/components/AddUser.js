import React, { Component } from 'react';
import { func } from 'prop-types';

class AddUser extends Component {
  state = {
    username: '',
    email: ''
  }

  static propTypes = {
    addUser: func
  };

  handleChange = (e) => {
    const elem = e.target;
    const stateObj = {};

    stateObj[elem.name] = elem.value;
    this.setState(() => (stateObj));
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.addUser({
      username: this.state.username,
      email: this.state.email
    });
  }

  render() {
    const { username, email } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <input
          onChange={this.handleChange}
          type="text"
          name="username"
          placeholder="Username"
          value={username}
          required
        />
        <input
          onChange={this.handleChange}
          type="email"
          name="email"
          placeholder="Email"
          value={email}
          required
        />
        <button type="submit">Submit</button>
      </form>
    );
  }
};

export default AddUser;