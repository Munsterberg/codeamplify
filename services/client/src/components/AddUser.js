import React, { Component } from 'react';
import { func } from 'prop-types';

class AddUser extends Component {
  state = {
    username: '',
    email: '',
    password: ''
  }

  static propTypes = {
    addUser: func
  };

  handleChange = (e) => {
    const elem = e.target;
    const stateObj = {};

    stateObj[elem.name] = elem.value;
    this.setState(() => (stateObj));
  }

  onSubmit = (e) => {
    e.preventDefault();
    this.props.addUser({
      username: this.state.username,
      email: this.state.email,
      password: this.state.password
    });
  }

  render() {
    const { username, email, password } = this.state;
    return (
      <form onSubmit={this.onSubmit}>
        <input
          onChange={this.handleChange}
          type="text"
          name="username"
          placeholder="Username"
          value={username}
          required
        />
        <input
          onChange={this.handleChange}
          type="email"
          name="email"
          placeholder="Email"
          value={email}
          required
        />
        <input
          onChange={this.handleChange}
          type="password"
          name="password"
          placeholder="Password"
          value={password}
          required
        />
        <button type="submit">Submit</button>
      </form>
    );
  }
};

export default AddUser;