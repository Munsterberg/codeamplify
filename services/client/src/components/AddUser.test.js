import React from 'react';
import { shallow, mount } from 'enzyme';

import AddUser from './AddUser';

describe('<AddUser />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AddUser />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  it('should render form', () => {
    expect(wrapper.find('form').length).toBe(1);
  });

  it('should contain proper inputs', () => {
    const elem = wrapper.find('form');
    expect(elem.find('input').get(0).props.name).toBe('username');
    expect(elem.find('input').get(1).props.name).toBe('email');
    expect(elem.find('button').length).toBe(1);
  });

  it('should call handleChange', () => {
    const instance = wrapper.instance();
    instance.handleChange({ target: { name: 'username', value: 'myuser' } });
    expect(wrapper.state().username).toBe('myuser');
  });

  it('should call onSubmit', () => {
    const addUser = jest.fn();
    const preventDefault = jest.fn();
    wrapper = shallow(<AddUser addUser={addUser} />);
    wrapper.setState({ username: 'test', email: 'test@test.com' });
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(addUser).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });
});