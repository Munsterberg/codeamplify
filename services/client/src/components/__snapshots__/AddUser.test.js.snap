// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> should call onSubmit 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="username"
    onChange={[Function]}
    placeholder="Username"
    required={true}
    type="text"
    value="test"
  />
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="email"
    value="test@test.com"
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`<AddUser /> should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="username"
    onChange={[Function]}
    placeholder="Username"
    required={true}
    type="text"
    value=""
  />
  <input
    name="email"
    onChange={[Function]}
    placeholder="Email"
    required={true}
    type="email"
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
